### DISCUSSION TITLE
Require one out of multiple sets of checkboxes

### DISCUSSION THREAD

**Discussion by ThinkSalat** [react-hook-form/react-hook-form#2523] [Q&A] [answered]
Hi,

I have multiple different sets of checkboxes, each with different names. I want to require just one of the checkboxes in any of the sets to be checked. I can't figure out a way to do this and would really appreciate some guidance on this seemingly simple task!

Thanks

**Comment by bluebill1049** [comment]
hey @ThinkSalat i think you use `validate` function with `getValues` right? and do a simple function to check them all checked. let me know if this helps.

**Reply by ThinkSalat** [reply]
Interesting.  I ended up refactoring them to all be named the same, then filtered them into correct arrays in onsubmit. 

I was using validate `watched` values and seeing if any arrays had any values. This wasn't working as the validation was triggering using the previous render's arrays.

**Comment by ThinkSalat** [comment, marked as answer]
If anyone is trying to figure out a workaround, I ended up refactoring them to all be named the same, then filtered them into correct arrays in onsubmit. This is kind hacky though and perhaps bluebill's suggestion would work as well.
